### PR TITLE
update docs for vue 3 component tracking update

### PR DESCRIPTION
**Pull request by snoozbuster**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [x] Checked Vercel preview for correctness, including links
- [x] PR was reviewed and approved by any necessary SMEs
- [x] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

Adds documentation for Vue 3 compatible performance tracing hooks added in https://github.com/getsentry/sentry-javascript/pull/9578. Screenshot of changed section:

image

I chose to update the examples to be vue 3 first because vue 2 is [reaching EOL](https://v2.vuejs.org/lts/) at the end of the year - so it made sense to update the docs here to be Vue 3 first as that is already the default for most people right now (or at least, I would expect so).

## Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by shanamatthews** [comment]
Hey @Lms24 - could you double check this is all accurate? Also, we need to wait until the next release to merge this, yeah?

File: src/platforms/javascript/guides/vue/features/component-tracking.mdx
Comment (shanamatthews) [inline src/platforms/javascript/guides/vue/features/component-tracking.mdx:66]:
```suggestion
In Vue 2, use `destroy` instead of `unmount`. `destroy` does not work in Vue 3, as the names of the lifecycle hooks themselves [changed](https://v3-migration.vuejs.org/breaking-changes/#other-minor-changes) in Vue 3.
```

**Review by shanamatthews — APPROVED** [review]
Looks good from the docs perspective!

**Review by Lms24 — APPROVED** [review]
Hey @snoozbuster thanks a lot for updating docs - appreciate your help! I agree we should show Vue 3 hooks more prominently. Makes sense to me. 

@shanamatthews thanks for the ping! We can merge this anytime. The change was already released in [7.81.0 ](https://github.com/getsentry/sentry-javascript/releases/tag/7.81.0)

**Pull request merged**
